components/supervisor/pkg/supervisor/tasks.go

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) 2020 TypeFox GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package supervisor
+
+import (
+	"bufio"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	csapi "github.com/gitpod-io/gitpod/content-service/api"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/gitpod-io/gitpod/supervisor/pkg/backup"
+	"github.com/gitpod-io/gitpod/supervisor/pkg/terminal"
+)
+
+type tasksManager struct {
+	config            *Config
+	tasks             map[string]*api.TasksStatus
+	ready             chan struct{}
+	terminalService   *terminal.MuxTerminalService
+	inWorkspaceHelper *backup.InWorkspaceHelper
+}
+
+func newTasksManager(config *Config, terminalService *terminal.MuxTerminalService, inWorkspaceHelper *backup.InWorkspaceHelper) *tasksManager {
+	return &tasksManager{
+		config:            config,
+		terminalService:   terminalService,
+		inWorkspaceHelper: inWorkspaceHelper,
+		tasks:             make(map[string]*api.TasksStatus),
+		ready:             make(chan struct{}),
+	}
+}
+
+func (tm *tasksManager) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	defer close(tm.ready)
+
+	tasks, err := tm.config.getGitpodTasks()
+	if err != nil {
+		log.WithError(err).Fatal()
+		return
+	}
+	if tasks == nil {
+		log.Info("no gitpod tasks found")
+		return
+	}
+
+	select {
+	case <-ctx.Done():
+		return
+	case <-tm.inWorkspaceHelper.ContentReady():
+		contentSource, _ := tm.inWorkspaceHelper.ContentSource()
```

Comment:
You could leave the `<-tm.inWorkspaceHelper.ContentReady():` case empty, thereby just continuing the flow. This way you'd need one indentation level less.